### ISSUE TITLE
`axes` for `conv` of `OffsetArrays` seems off

### ISSUE THREAD

**Issue by jishnub** [JuliaDSP/DSP.jl#504]
```julia
julia> a = 1:2
1:2

julia> conv(a, a)
3-element Vector{Int64}:
 1
 4
 4

julia> conv(OffsetArray(a), OffsetArray(a))
3-element OffsetArray(::Vector{Int64}, 2:4) with eltype Int64 with indices 2:4:
 1
 4
 4

julia> a == OffsetArray(a)
true

julia> conv(a, a) == conv(OffsetArray(a), OffsetArray(a))
false
```
I'm unsure what's the axis logic here, but it'll be good for this to be consistent.

**Comment by martinholters** [comment]
Yeah, the issue is that convolution is naturally defined with zero-based indexing, i.e.
```
y[n] = sum(a[n-i] * b[i] for i in ...)
```
And that's what you get for `OffsetArray`. But as plain `Array`s are one-based, this would always give a leading zero in the result, which would be highly undesirable. So whatever one does, it will always be inconsistent in some way.

Maybe we'd need a keyword argument `zero_based` or the like to opt into the behavior currently adopted automatically for `OffsetArray`s? Introducing it would be breaking if not done carefully, though.
